ps_facetedsearch.php

```diff
@@ -647,8 +629,15 @@ public function getContent()
                     }
 
                     $this->buildLayeredCategories();
-                    $message = $this->displayConfirmation($this->trans('Your filter', [], 'Modules.Facetedsearch.Admin') . ' "' . Tools::safeOutput(Tools::getValue('layered_tpl_name')) . '" ' .
-                        ((isset($_POST['id_layered_filter']) && $_POST['id_layered_filter']) ? $this->trans('was updated successfully.', [], 'Modules.Facetedsearch.Admin') : $this->trans('was added successfully.', [], 'Modules.Facetedsearch.Admin')));
+                    $message = $this->displayConfirmation(
+                        $this->trans('Your filter', [], 'Modules.Facetedsearch.Admin') . ' "' .
+                        Tools::safeOutput(Tools::getValue('layered_tpl_name')) . '" ' .
+                        (
+                            !empty($_POST['id_layered_filter']) ?
+                            $this->trans('was updated successfully.', [], 'Modules.Facetedsearch.Admin') :
+                            $this->trans('was added successfully.', [], 'Modules.Facetedsearch.Admin')
+                        )
+                    );
```

Comment:
For RTL languages, consider having one `trans` having the whole sentence. For instance

```php
$resultString = Tools::safeOutput(Tools::getValue('layered_tpl_name')) . '" ' .
                        (
                            !empty($_POST['id_layered_filter']) ?
                            $this->trans('was updated successfully.', [], 'Modules.Facetedsearch.Admin') :
                            $this->trans('was added successfully.', [], 'Modules.Facetedsearch.Admin')
                        );

$message = $this->displayConfirmation(
    $this->trans('Your filter %s', ['%resultString%' => $resultString], 'Modules.Facetedsearch.Admin')
);
```

Reply:
it's the legacy translations and because they are in native modules I don't know if I can change it :/ @LouiseBonnard wdyt?

Reply:
Oh they were already here in the previous version. My bad.

Reply:
Hi there! @Quetzacoalt91 is right, those strings are used in this module since (at least) the 1.7.2 version. Do you still need to change it?

Reply:
I'm not changing it. He asked it because the current translating behavior is weird here. And his suggestion was legit.
Can we plan to refactor translations in the future?

Reply:
I do hope so, but not before the next major version.

Reply:
This conversation is solved. No change required